vendor/github.com/openshift/library-go/pkg/manifest/manifest.go

```diff
@@ -114,11 +116,46 @@ func (m *Manifest) UnmarshalJSON(in []byte) error {
 	return validateResourceId(m.id)
 }
 
+const featureSetAnnotation = "release.openshift.io/feature-set"
+
+var knownFeatureSets = sets.String{}
+
+func init() {
+	for featureSet := range configv1.FeatureSets {
+		if len(featureSet) == 0 {
+			knownFeatureSets.Insert("Default")
+			continue
+		}
+		knownFeatureSets.Insert(string(featureSet))
+	}
+}
+
+func getFeatureSets(annotations map[string]string) (sets.String, bool, error) {
+	ret := sets.String{}
+	specified := false
+	for _, featureSetAnnotation := range []string{"release.openshift.io/feature-gate", "release.openshift.io/feature-set"} {
+		featureSetAnnotationValue, featureSetAnnotationExists := annotations[featureSetAnnotation]
+		if featureSetAnnotationExists {
+			specified = true
+			featureSetAnnotationValues := strings.Split(featureSetAnnotationValue, ",")
+			for _, manifestFeatureSet := range featureSetAnnotationValues {
+				if !knownFeatureSets.Has(manifestFeatureSet) {
+					// never include the manifest if the feature-set annotation is outside of allowed values (only TechPreviewNoUpgrade and "" are currently allowed)
```

Comment:
Can we drop `(only TechPreviewNoUpgrade and "" are currently allowed)`, because I expect it would go stale, and may already be stale with your `configv1.FeatureSets` source for `knownFeatureSets`.